### PR TITLE
Show correct copy on finish page

### PR DESCRIPTION
**Pull request by shanear**
connects #3437

### CODE REVIEW

File: client/app/intake/pages/finish.jsx [hunk 1]
```diff
@@ -21,16 +21,27 @@ class Finish extends React.PureComponent {
     default:
     }
 
+    let ep, optionName;
```
Comment (nickheiner-usds) [inline client/app/intake/pages/finish.jsx]: how about `epName` for more descriptiveness?

File: client/app/intake/pages/finish.jsx [hunk 2]
```diff
@@ -21,16 +21,29 @@ class Finish extends React.PureComponent {
...
-      <span>Update the Subject Line with <em>RAMP Opt-In</em>.</span>,
-      <span>Create an EP <strong>030 RAMP Supplemental</strong> in VBMS.</span>,
-      <span>Add a placeholder contention of <em>RAMP</em>.</span>
+      <span>Upload the RAMP Election form to the VBMS eFolder and ensure the Document Type is <b>Correspondence</b>.</span>,
+      <span>Update the Subject Line with "Ramp Election."</span>,
+      <span>Create an EP <strong>{ epName }</strong> in VBMS.</span>,
+      <span>Add a placeholder contention of "RAMP."</span>,
+      <span>Send a RAMP Withdrawal Letter using Letter Creator.</span>
     ];
-    const stepFns = steps.map((step, index) => () => <span><strong>Step {index}.</strong> {step}</span>);
```
Comment (nickheiner-usds) [inline client/app/intake/pages/finish.jsx]: Why was this change made?

**Comment by NickHeiner** [comment]
image

Ha, I didn't notice this before, but we probably shouldn't start with step `0`. :smile:

File: client/app/intake/pages/finish.jsx [hunk 3]
```diff
@@ -21,16 +21,29 @@ class Finish extends React.PureComponent {
...
     const steps = [
-      <span>Upload the RAMP election form to VBMS and ensure the Document Type is <em>Correspondence</em>.</span>,
-      <span>Update the Subject Line with <em>RAMP Opt-In</em>.</span>,
-      <span>Create an EP <strong>030 RAMP Supplemental</strong> in VBMS.</span>,
-      <span>Add a placeholder contention of <em>RAMP</em>.</span>
+      <span>Upload the RAMP Election form to the VBMS eFolder and ensure the Document Type is <b>Correspondence</b>.</span>,
+      <span>Update the Subject Line with "Ramp Election."</span>,
+      <span>Create an EP <strong>{ epName }</strong> in VBMS.</span>,
+      <span>Add a placeholder contention of "RAMP."</span>,
+      <span>Send a RAMP Withdrawal Letter using Letter Creator.</span>
```
Comment (NickHeiner) [inline client/app/intake/pages/finish.jsx]: The spec calls for `Letter Creator` to be italicized.

**Comment by NickHeiner** [comment]
image

The spec calls for "steps", not "4 steps". And right now we have 5 steps. So let's just remove the count from the sentence.

File: client/app/intake/pages/finish.jsx (hunk 2 above)
Comment (shanear) [reply client/app/intake/pages/finish.jsx]: line length violation

File: spec/feature/intake_spec.rb [hunk 4]
```diff
@@ -79,6 +79,36 @@
       expect(intake).to be_canceled
     end
 
+    scenario "Start intake and go back and edit option" do
+      Generators::Appeal.build(vbms_id: "12341234C", vacols_record: :ready_to_certify)
```
Comment (NickHeiner) [inline spec/feature/intake_spec.rb]: Is the `vbms_id` always supposed to be `veteran_file_number + 'C'`?

File: spec/feature/intake_spec.rb [hunk 5]
```diff
@@ -103,21 +133,21 @@
       )
 
       within_fieldset("Which election did the Veteran select?") do
-        find("label", text: "Supplemental Claim").click
```
Comment (NickHeiner) [inline spec/feature/intake_spec.rb]: Why did we make this change? (And the changes below in this file.)

File: spec/feature/intake_spec.rb [hunk 6]
```diff
@@ -79,6 +79,36 @@
...
+      intake.start!
+
+      # Validate that visiting the finish page takes you back to
+      # the review request page if you haven't yet reviewed the intake
+      visit "/intake/completed"
+
+      within_fieldset("Which election did the Veteran select?") do
+        find("label", text: "Higher Level Review without DRO hearing request").click
+      end
+      fill_in "What is the Receipt Date for this election form?", with: "08/07/2017"
```
Comment (NickHeiner) [inline spec/feature/intake_spec.rb]: Why do we need to fill in the date? (I do see that it doesn't show up in the UI.) If we're setting `notice_date` above, shouldn't it automatically populate on the frontend?

File: spec/feature/intake_spec.rb (hunk 4 above)
Comment (shanear) [reply spec/feature/intake_spec.rb]: not always, look here if you want to know:
https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/appeal.rb#L569

File: spec/feature/intake_spec.rb (hunk 5 above)
Comment (shanear) [reply spec/feature/intake_spec.rb]: to cover all the options.

File: spec/feature/intake_spec.rb (hunk 6 above)
Comment (shanear) [reply spec/feature/intake_spec.rb]: this is the receipt date

**Comment by NickHeiner** [comment]
I see that the tests fail when the code is broken.

I see that the copy is correct with what #3437 specifies, aside from comments noted above.

File: client/app/intake/pages/finish.jsx (hunk 2 above)
Comment (NickHeiner) [reply client/app/intake/pages/finish.jsx]: With the current update, I see it as `110` chars. Our limit is `120`.

File: client/app/intake/pages/finish.jsx [hunk 7]
```diff
@@ -75,6 +91,7 @@ export class FinishButtons extends React.PureComponent {
 
 export default connect(
   (state) => ({
+    rampElection: state.rampElection,
```
Comment (NickHeiner) [inline client/app/intake/pages/finish.jsx]: It's not worth doing another roundtrip to fix now, but generally, you want to pull the absolute minimum amount of information off of `state`. By default, React re-renders a component if anything in that component's props change. The only field we use from `rampElection` is `optionSelected`, but if anything else on `rampElection` changes, we'd have a wasted render.

File: spec/feature/intake_spec.rb (hunk 6 above)
Comment (NickHeiner) [reply spec/feature/intake_spec.rb]: image

File: client/app/intake/pages/finish.jsx (hunk 7 above)
Comment (shanear) [reply client/app/intake/pages/finish.jsx]: 👍 thanks for explaining that, i'll make a note to clean it all up

**Pull request merged**